parking_permits_app/customer_permit.py

```diff
@@ -164,16 +169,26 @@ def update(self, data, permit_id=None):
                 }
             )
             if permit_id:
-                self.customer_permit_query.filter(id__in=permit_to_update).update(
-                    **fields_to_update
-                )
-                return self.customer_permit_query.get(id=permit_id)
+                permit = self.customer_permit_query.get(id=permit_id)
+                return self._add_reversion(permit, fields_to_update)
 
         return self._update_fields_to_all_draft(fields_to_update)
 
     def _update_fields_to_all_draft(self, data):
-        self.customer_permit_query.filter(status=DRAFT).update(**data)
-        return self.customer_permit_query.all()
+        permits = self.customer_permit_query.filter(status=DRAFT).all()
+        return [self._add_reversion(permit, data) for permit in permits]
+
+    def _add_reversion(self, permit, data):
+        keys = data.keys()
+        with reversion.create_revision():
+            for key in keys:
+                setattr(permit, key, data[key])
```

Comment:
Not blocking the merge, but you could do something like this without extracting keys first:
```suggestion
            for key, value in data.items():
                setattr(permit, key, value)
```

Reply:
What about `permit.save(update_fields=keys)`?